### PR TITLE
homepage modal, stimulus add fishing trip

### PR DESCRIPTION
**Pull request by jweber13**

### CODE REVIEW

**Comment by jweber13** [comment]
Merge Conflicts with controllers / index.js & Yarn.lock. 

Also, need to re-write the modal to close upon success. it will close with cancel, and 'submit' does add the name to the db & create a new fishing  trip.

File: app/views/pages/home.html.erb [hunk 1]
```diff
@@ -1,2 +1,65 @@
...
+            <line x1="6" y1="6" x2="18" y2="18" />
+          </svg>
+        </button>
+      </div>
+    </div>
+  </div>
+  <div class="w-full flex justify-center py-12" id="button">
+    <button class="focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-700 mx-auto transition duration-150 ease-in-out hover:bg-indigo-600 bg-indigo-700 rounded text-white px-4 sm:px-8 py-2 text-xs sm:text-sm" onclick="modalHandler(true)">Go Fishin'</button>
+  </div>
+  <script>
```
Comment (Donald-Chow) [inline app/views/pages/home.html.erb]: Why are there Javascript in the ERB?!?!??!?!?!

File: app/javascript/controllers/index.js [hunk 2]
```diff
@@ -4,5 +4,11 @@
 
 import { application } from "./application"
 
+import FishingTripInputController from "./fishing_trip_input_controller"
+application.register("fishing-trip-input", FishingTripInputController)
+
 import HelloController from "./hello_controller"
 application.register("hello", HelloController)
+
```
Comment (Donald-Chow) [inline app/javascript/controllers/index.js]: This controller is no longer valid

! Remove

File: app/policies/application_policy.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 class ApplicationPolicy
   attr_reader :user, :record
+  include Devise::Controllers::Helpers
```
Comment (Donald-Chow) [inline app/policies/application_policy.rb]: Are we still using Devise in Pundit?
If not, ! Remove

File: app/views/pages/home.html.erb (hunk 1 above)
Comment (Donald-Chow) [reply app/views/pages/home.html.erb]: Looks like this is the DOM changes on click.
! move to stimulus controller

**Review by Donald-Chow — CHANGES_REQUESTED** [review]
Clean up Javascript before pull

Index is a conflict in the controller, we should accept both
Yarn Lock conflict should be on versioning, suggest to keep base version of yarn lock when merge

**Comment by Donald-Chow** [comment]
@jweber13 I've made a bunch of changes to the stimulus controllers (both modalhandler and fishing-trip-input) and adjusted the syntax accordingly. In addition, I have made the modal a partial where we can render from anywhere.

**Please review**

Additional consideration, if we use simpleform, if the trip is created unsuccessfully, an alert will be created to tell the user why the trip is not created properly. if that is something we want to do. alternative/ additional, we can add flashes when the trip is created properly

**Comment by Donald-Chow** [comment]
@jweber13 I just merged master and resolve conflict.

Please review

**Review by Donald-Chow — APPROVED** [review]

**Pull request merged**